src/event/event.rs

```diff
@@ -71,6 +82,17 @@ impl Event {
     /// Because of the above be cautions when using this in cross-platform
     /// applications, Mio makes no attempt at normalising this indicator and
     /// only provides a convenience method to read it.
+    ///
+    /// The table below shows what flags are checked on what OS.
+    ///
+    /// | [OS selector] | Flag(s) checked |
+    /// |---------------|-----------------|
+    /// | [epoll]       | `EPOLLHUP` and `EPOLLRDHUP` |
```

Comment:
for `hup` **only** `EPOLLHUP` can be checked (nothing for kqueue).

We need to add a separate `rdhup` event, which can be supported by all OSes (tracking issue #1040).